src/DbUtils.php

```diff
@@ -537,13 +541,17 @@ public function countDistinctElementsInTable($table, $field, $condition = [])
     {
 
         if (!is_array($condition)) {
+            Toolbox::Deprecated('Condition must be an array!');
             if (empty($condition)) {
                 $condition = [];
             }
         }
-        $condition['COUNT'] = 'cpt';
-        $condition['FIELDS'] = $field;
-        $condition['DISTINCT'] = true;
+
+        if (is_array($condition)) {
+            $condition['COUNT'] = 'cpt';
+            $condition['FIELDS'] = $field;
+            $condition['DISTINCT'] = true;
```

Comment:
I guess that if `$condition` is not an array, then the `DISTINCT` keyword will probably not be present in the query, so result will not be valid.